.travis.yml

```diff
@@ -35,22 +36,20 @@ after_failure:
         - if [ -s _build/mim2/rel/mongooseim/log/crash.log ]; then cat _build/mim2/rel/mongooseim/log/crash.log; fi
         - tail -100 _build/mim1/rel/mongooseim/log/ejabberd.log
         - tail -100 _build/mim2/rel/mongooseim/log/ejabberd.log
+        - tools/travis-prepare-log-dir.sh
+        - if [ -n "${AWS_SECRET_ACCESS_KEY}" ]; then tools/travis-upload-to-s3.sh; fi
+        - if [ -n "${GDRIVE_SERVICE_ACCOUNT_CREDENTIALS}" ]; then tools/travis-upload-to-gdrive.sh; fi
+        - tools/travis-secure-erase.sh /tmp/serviceAccountCredentials
 
 after_success:
         # env variables help to distinguish CI builds
         # It takes less time without gcov
-        - codecov --disable=gcov --env PRESET
-        - ./rebar3 coveralls send
+        - test 1 = "$SKIP_COV" || codecov --disable=gcov --env PRESET
+        - test 1 = "$SKIP_COV" || ./rebar3 coveralls send
         - if [ -n "${DOCKERHUB_PASS}" ] && [ $PRESET = 'internal_mnesia' ] && [ $TRAVIS_OTP_RELEASE = "19.3" ];
             then tools/travis-build-and-push-docker.sh;
           fi
 
-after_script:
-        - tools/travis-prepare-log-dir.sh
-        - if [ -n "${AWS_SECRET_ACCESS_KEY}" ]; then tools/travis-upload-to-s3.sh; fi
```

Comment:
We've been here some time ago. In my opinion it's needed to upload results to s3 in any case. I remember a situation when we needed to access old results from successful jobs, but they weren't there. That's why since then test results are uploaded after every job (failed or not).

Reply:
Can we upload it when:
- branch is master
- or build is failing?

Because I don't want to wait 3 minutes per task for PR builds.

Reply:
reverted for now.
Any ideas how to optimize uploading is a topic for a separate PR.

Reply:
>
>Can we upload it when:
>  * branch is master
>  * or build is failing?
> Because I don't want to wait 3 minutes per task for PR builds.

I think it's enough to upload them for master branch build, tag builds and all failed builds.